lib/iris/fileformats/netcdf.py

```diff
@@ -1973,19 +1968,26 @@ def store(data, cf_var, fill_value):
 
         # If packing attributes are specified, don't bother checking whether
         # the fill value is in the data.
-        fill_value_to_check = None if packing else fill_value
+        fill_value_to_check = None if packing else \
+            fill_value if fill_value is not None else \
+            netCDF4.default_fillvals[dtype.str[1:]]
 
         # Store the data and check if it is masked and contains the fill value
         is_masked, contains_fill_value = store(data, cf_var,
                                                fill_value_to_check)
 
-        if is_masked:
-            if contains_fill_value:
-                warnings.warn("Cube '{}' contains fill value {}. Some data "
-                              "points will be masked.".format(cube.name(),
-                                                              fill_value))
-        else:
-            cf_var.delncattr('_FillValue')
+        if dtype.itemsize == 1 and fill_value is None:
+            if is_masked:
+                warnings.warn("Cube '{}' contains masked byte data and will "
+                              "be interpreted as unmasked. To save as masked "
+                              "data please explicitly provide a fill value."
+                              .format(cube.name()))
+        elif contains_fill_value:
+            warnings.warn("Cube '{}' contains data points equal to the fill"
+                          "value {}. The points will be interpreted as being "
+                          "masked. Please provide a fill_value argument not "
+                          "equal to any data point.".format(cube.name(),
```

Comment:
@pp-mo Please feel free to suggest a rewording of these warnings

Reply:
I will consider this later, as I'm thinking to integrate this with the same requirements in the PP saver
( #2744, to be updated when this is done )